contentcuration/contentcuration/frontend/channelEdit/views/trash/__tests__/trashModal.spec.js

```diff
@@ -1,9 +1,44 @@
-import { mount } from '@vue/test-utils';
+import { mount, createLocalVue } from '@vue/test-utils';
+import Vuex, { Store } from 'vuex';
 import VueRouter from 'vue-router';
 import TrashModal from '../TrashModal';
-import store from '../../../store';
 import { RouterNames } from '../../../constants';
 
+const localVue = createLocalVue();
+localVue.use(Vuex);
+localVue.use(VueRouter);
+
+const currentChannelGetters = {
+  currentChannel: jest.fn(),
+  trashId: jest.fn(),
+};
+const contentNodeGetters = {
+  getContentNodeChildren: jest.fn(),
+};
+const contentNodeActions = {
+  deleteContentNodes: jest.fn(),
+  loadTrashTree: jest.fn(),
+  loadContentNodes: jest.fn(),
+};
+const contentNodeMutations = {
+  SET_MOVE_NODES: jest.fn(),
+};
+
+const store = new Store({
```

Comment:
This was needed as all tests would fail because of
`throw new Error('[loadTrashTree] Channel data are not loaded');` below
as channel is not provisioned.

It would be better to revisit larger views tests to use mocked actions and getters data instead of reaching into internals and mocking computed properties as here though this is out of scope of this PR so I've just mocked it with empty values to make it pass and will open an issue for us to agree on some general testing strategy for larger views.

Reply:
The trash modal needs updated I believe which is probably why this needs updated.

Reply:
I am not sure if I follow here, could you elaborate what kind of update trash modal needs? Trash modal tests don't take it into account at all at this point but preferably they should though I don't want to work on this here. `TrashModal` component/route is child of `TreeView` route (see router.js) so channel data should be there, right?

Reply:
If you check the spec file in detail, for tests only mocked computed properties are currently being used.